packages/dd-trace/src/proxy.js

```diff
@@ -26,6 +26,15 @@ class Tracer extends NoopProxy {
     try {
       const config = new Config(options) // TODO: support dynamic config
 
+      const isDeprecatedGCPFunction = process.env.FUNCTION_NAME !== undefined && process.env.GCP_PROJECT !== undefined
+      const isNewerGCPFunction = process.env.K_SERVICE !== undefined && process.env.FUNCTION_TARGET !== undefined
+      const inGCPFunction = isDeprecatedGCPFunction || isNewerGCPFunction
```

Comment:
we have to check isGCPFunction within proxy.js as requiring + calling `maybeStartServerlessMiniAgent` takes 7ms (even when `maybeStartServerlessMiniAgent` just returns immediately)